### PR TITLE
feat(bootstrap): brew

### PR DESCRIPTION
**Pull request by joshuarli**
This installs brew and sentry's brew dependencies.

### CODE REVIEW

File: lib/brew.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+from devenv.lib import fs
+from devenv.lib import proc
+
+repo_path = "/opt/homebrew"
+
+
+def install() -> None:
+    if which("brew") == f"{repo_path}/bin/brew":
+        return
```
Comment (mattgauntseo-sentry) [inline lib/brew.py:14]: Can you add a comment about this check.

I.e. does this mean than brew bundle has been run before?

File: lib/brew.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    shellrc = fs.shellrc()
+    print("You may be prompted for your password to install homebrew.")
+    while True:
```
Comment (mattgauntseo-sentry) [inline lib/brew.py:18]: Should we limit how many times this runs?

File: lib/brew.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+import os
+from shutil import which
+
+from devenv.lib import fs
+from devenv.lib import proc
+
+repo_path = "/opt/homebrew"
```
Comment (mattgauntseo-sentry) [inline lib/brew.py:9]: Can you rename this to something like "brew_install_path". I started to think we were installing brew into the sentry repo

**Review by mattgauntseo-sentry — APPROVED** [review]

File: lib/brew.py (hunk 1 above)
Comment (joshuarli) [reply lib/brew.py:14]: done. no, it's just concerned with installing brew

File: lib/brew.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+""",
+                ),
+                exit=False,
+            )
+        except RuntimeError:
+            continue
+        break
+
+    if not os.path.exists(f"{repo_path}/bin/brew"):
+        os.makedirs(repo_path, exist_ok=True)
```
[the remark this answers is not in the document]
Comment (joshuarli) [reply lib/brew.py:37]: ah you're right, sorry this is an artifact before i added the which("brew") in the top

File: lib/brew.py (hunk 2 above)
Comment (joshuarli) [reply lib/brew.py:18]: No, people can ^C out at any time.

**Pull request merged**